skxray/diff_roi_choice.py

```diff
@@ -360,18 +383,83 @@ def _process_rings(num_rings, img_dim, ring_vals, ring_inds):
         pixel indices for the required rings
 
     """
-    # find the pixel list
-    w = np.where(ring_inds > 0)
-    grid = np.indices((img_dim[0], img_dim[1]))
-    pixel_list = (grid[0]*img_dim[1] + grid[1]).flatten()[w]
+    mesh = np.zeros(detector_size, dtype=np.int64)
+    angles = np.linspace(0, 360, num_angles)
 
-    ring_inds = ring_inds[ring_inds > 0]
+    yy, xx = np.mgrid[:detector_size[0], :detector_size[1]]
+    y_ = (np.flipud(yy) - calibrated_center[1])
+    x_ = (xx - calibrated_center[0])
+    grid_values = np.float_(np.hypot(x_, y_))
+    angle_grid = np.arctan2(y_, x_)*180/(np.pi)
 
-    ring_vals = np.array(ring_vals)
-    ring_vals = ring_vals.reshape(num_rings, 2)
+    angle_grid[grid_values > radius] = np.nan
+
+    for i in range(num_angles-1):
```

Comment:
I think if you do this an `np.floor(angle_grid / (2*np.pi / num_angles)))` you can then use `bin_count` again.